MSAL/src/MSALInteractiveTokenParameters.m

```diff
@@ -28,21 +28,73 @@
 #import "MSALInteractiveTokenParameters.h"
 #import "MSALTokenParameters+Internal.h"
 #import "MSALGlobalConfig.h"
+#import "MSALWebviewParameters.h"
 
 @implementation MSALInteractiveTokenParameters
 
 @synthesize telemetryApiId;
 
 - (instancetype)initWithScopes:(NSArray<NSString *> *)scopes
+             webviewParameters:(MSALWebviewParameters *)webviewParameters
 {
     self = [super initWithScopes:scopes];
     if (self)
     {
-        self.webviewType = MSALGlobalConfig.defaultWebviewType;
-        self.promptType = MSALPromptTypeDefault;
         self.telemetryApiId = MSALTelemetryApiIdAcquireWithTokenParameters;
+        _promptType = MSALPromptTypeDefault;
+        _webviewParameters = [webviewParameters copy];
     }
     return self;
 }
 
+- (instancetype)initWithScopes:(NSArray<NSString *> *)scopes
+{
+    return [self initWithScopes:scopes
+                  webviewParameters:[MSALWebviewParameters new]];
+}
+
+#if TARGET_OS_IPHONE
+
+- (void)setParentViewController:(UIViewController *)parentViewController
```

Comment:
are we sure this won't be needed for macOS?

Reply:
Not sure, but [keyWindow](https://developer.apple.com/documentation/appkit/nsapplication/1428406-keywindow) isn't deprecated for MacOS :) We will need to double test this when we start working on MacOS changes.